tests/test_interfaces.py

```diff
@@ -420,3 +420,223 @@ def test_output_format(
+
+
+# Tests to assert that incorrect input to calibrate is caught
+@pytest.mark.parametrize("model_fixture", MODELS)
+def test_positive_int_calibrate(model_fixture):
+    # Assert that providing a positive int does NOT raise an error
+    if model_fixture.data_path is None or model_fixture.data_mapping is None:
+        pytest.skip("Skip models with no data attached")
+    calibrate(
+        model_fixture.url,
+        model_fixture.data_path,
+        data_mapping=model_fixture.data_mapping,
+        num_iterations=3,
+    )
+
+
+@pytest.mark.parametrize("model_fixture", MODELS)
+def test_float_calibrate(model_fixture):
+    # Assert that providing a float raises a ValueError
+    if model_fixture.data_path is None or model_fixture.data_mapping is None:
+        pytest.skip("Skip models with no data attached")
+    with pytest.raises(ValueError):
+        calibrate(
+            model_fixture.url,
+            model_fixture.data_path,
+            data_mapping=model_fixture.data_mapping,
+            num_iterations=3.5,
+        )
+
+
+@pytest.mark.parametrize("model_fixture", MODELS)
+def test_negative_int_calibrate(model_fixture):
+    # Assert that providing a negative integer raises a ValueError
+    if model_fixture.data_path is None or model_fixture.data_mapping is None:
+        pytest.skip("Skip models with no data attached")
+    with pytest.raises(ValueError):
+        calibrate(
+            model_fixture.url,
+            model_fixture.data_path,
+            data_mapping=model_fixture.data_mapping,
+            num_iterations=-3,
+        )
+
+
+@pytest.mark.parametrize("model_fixture", MODELS)
+def test_zero_calibrate(model_fixture):
+    # Assert that providing 0 raises a ValueError
+    if model_fixture.data_path is None or model_fixture.data_mapping is None:
+        pytest.skip("Skip models with no data attached")
+    with pytest.raises(ValueError):
+        calibrate(
+            model_fixture.url,
+            model_fixture.data_path,
+            data_mapping=model_fixture.data_mapping,
+            num_iterations=0,
+        )
+
+
+@pytest.mark.parametrize("model_fixture", MODELS)
+def test_tensor_int_calibrate(model_fixture):
+    # Assert that providing a tensor with an int raises a ValueError
+    if model_fixture.data_path is None or model_fixture.data_mapping is None:
+        pytest.skip("Skip models with no data attached")
+    with pytest.raises(ValueError):
+        calibrate(
+            model_fixture.url,
+            model_fixture.data_path,
+            data_mapping=model_fixture.data_mapping,
+            num_iterations=torch.tensor(5),
+        )
+
+
+# Tests to assert that incorrect input to sample is caught
+@pytest.mark.parametrize("model_fixture", MODELS)
+@pytest.mark.parametrize("end_time", END_TIMES)
+@pytest.mark.parametrize("logging_step_size", LOGGING_STEP_SIZES)
+def test_positive_int_sample(model_fixture, end_time, logging_step_size):
```

Comment:
For the `sample` and `ensemble_sample` tests, could you consolidate them into parameterized tests using `SAMPLE_METHODS`. See https://github.com/ciemss/pyciemss/pull/465/files#diff-fe07557ea070948dd5e37c1b32b8f92dbb6a87d96d98a297f6d994db44a7cc6fR399 for an example.

For example, `test_positive_int_sample` would take as input an argument `sample_method` and then call `sample_method` in place of `sample`.

Reply:
Fixed this as well, also changed from using `model_fixture` as input to using just the urls.